app/src/main/java/com/eveningoutpost/dexdrip/Models/Libre2Sensor.java

```diff
@@ -0,0 +1,55 @@
+package com.eveningoutpost.dexdrip.Models;
+
+import android.provider.BaseColumns;
+import android.text.format.DateFormat;
+
+import com.activeandroid.annotation.Column;
+import com.activeandroid.annotation.Table;
+import com.activeandroid.query.Select;
+
+import java.sql.SQLException;
+import java.util.List;
+
+@Table(name = "Libre2Sensors", id = BaseColumns._ID)
+public class Libre2Sensor extends PlusModel {
+    static final String TAG = "Libre2Sensor";
+
+    static final String[] schema = {
+            "CREATE VIEW Libre2Sensors AS SELECT MIN(_id) as _id, serial, MIN(ts) as ts_from, MAX(ts) AS ts_to, COUNT(*) AS readings FROM Libre2RawValue2 GROUP BY serial ORDER BY ts DESC;"
+    };
+
+    @Column(name = "serial", index = true)
+    public String serial;
+
+    @Column(name = "ts_from", index = false)
+    public long ts_from;
+
+    @Column(name = "ts_to", index = false)
+    public long ts_to;
+
+    @Column(name = "readings", index = false)
+    public long readings;
+
+    public static String Libre2Sensors() {
```

Comment:
I am thinking this will create a database query every second when the megastatus display is in progress. How often does this data change? Would it be better to cache it in a static field in the class, for example:

```
private static volatile String cachedString = null;

if ((cachedString == null) || (JoH.ratelimit("libre2sensor-report",60)) {
...
cachedString = <result from calcs>;
}
return cachedString;
```

Reply:
Will check that, and test it. I come back after testing.

Reply:
@jamorham, I have set the query to 2 minutes as recommended. I tested it, and it works as expected.